vault/identity_store_oidc.go

```diff
@@ -1346,35 +1472,79 @@ func (i *IdentityStore) oidcKeyRotation(ctx context.Context, s logical.Storage)
 
 // oidcPeriodFunc is invoked by the backend's periodFunc and runs regular key
 // rotations and expiration actions.
-func (i *IdentityStore) oidcPeriodicFunc(ctx context.Context, s logical.Storage) {
-	nextRun := time.Time{}
+func (i *IdentityStore) oidcPeriodicFunc(ctx context.Context) {
+	var nextRun time.Time
+	now := time.Now()
 
-	if v, ok := i.oidcCache.Get("nextRun"); ok {
+	nsPaths := i.listNamespacePaths(ctx)
+
+	if v, ok := i.oidcCache.Get(nil, "nextRun"); ok {
 		nextRun = v.(time.Time)
 	}
 
 	// The condition here is for performance, not precise timing. The actions can
 	// be run at any time safely, but there is no need to invoke them (which
 	// might be somewhat expensive if there are many roles/keys) if we're not
 	// past any rotation/expiration TTLs.
-	if time.Now().After(nextRun) {
-		nextRotation, err := i.oidcKeyRotation(ctx, s)
-		if err != nil {
-			i.Logger().Warn("error rotating OIDC keys", "err", err)
-		}
+	if now.After(nextRun) {
+		// Initialize to a fairly distant next run time. This will be brought in
+		// based on key rotation times.
+		nextRun = now.Add(24 * time.Hour)
```

Comment:
What if rotation and expiration periods are greater than a day?

Reply:
It would just mean that we'd still run the _check_ (i.e. inspect whether things need to expire/rotate) at least once per day. The rotations and expirations always occur at the correct time (within the 1 minute periodicFunc). Since the checking involves listing and loading a bunch of keys, this was meant to just reduce that load.